rust-semantics/syntax.md

```diff
@@ -0,0 +1,1040 @@
+Partial Rust Syntax
+===================
+
+Macros create ambiguities, so this syntax would be cleaner if it could rely on
+macros being already expanded. However, expanding macros on MultiversX code
+would force us to support more Rust features in the main semantics.
+
+```k
+module RUST-SYNTAX
+    imports RUST-SHARED-SYNTAX
+
+    syntax Underscore ::= "_"  [token]
+
+```
+
+  https://doc.rust-lang.org/reference/identifiers.html
+
+```k
+
+  // TODO: Not implemented properly
+  syntax Identifier ::= r"[A-Za-z_][A-Za-z0-9\\_]*"  [prec(2), token]
+endmodule
+
+module RUST-SHARED-SYNTAX
+  imports STRING-SYNTAX
+```
+
+  https://doc.rust-lang.org/reference/crates-and-source-files.html
+
+```k
+    syntax Crate ::= InnerAttributes Items  [symbol(crate)]
+    syntax InnerAttributes ::= List{InnerAttribute, ""}  [symbol(innerAttributes)]
+    syntax Items ::= List{Item, ""}  [symbol(items)]
+
+```
+
+  https://doc.rust-lang.org/reference/attributes.html
+
+```k
+
+    syntax InnerAttribute ::= "#![" Attr "]"  [symbol(innerAttribute)]
+    syntax OuterAttribute ::= "#[" Attr "]"  [symbol(outerAttribute)]
+
+    syntax Attr ::= Identifier
```

Comment:
It seems, `Attr` syntax is defined with `SimplePath` (line 92), would you share the reason of not using `SimplePath` in the sort definition?